apps/tools/blocks-crawler/src/main.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use blocks::{check_block, fetch_block_info};
+use std::fs::create_dir_all;
+#[macro_use]
+extern crate log;
+
+mod blocks;
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+    // Create output folders
+    create_dir_all(&"output/succeeded/").unwrap();
+    create_dir_all(&"output/failed/").unwrap();
+
+    // Read limit value from cli
+    let limit = std::env::args().nth(1).unwrap();
+
+    // Fetch n blocks info from graphql api
+    let blocks = fetch_block_info(&limit).await.unwrap();
+    let mut handlers = vec![];
+    for block in blocks.data.blocks {
+        handlers.push(std::thread::spawn(move || {
```

Comment:
I doubt if this works with a large N.  Maybe refer to https://kerkour.com/rust-worker-pool#/io-intensive-jobs-async